core/src/main/java/io/grpc/ServerStreamTracer.java

```diff
@@ -54,4 +70,103 @@ public void serverCallStarted(ServerCall<?, ?> call) {
     public abstract ServerStreamTracer newServerStreamTracer(
         String fullMethodName, Metadata headers);
   }
+
+  /**
+   * A data class with info about the started {@link ServerCall}.
+   */
+  public static final class ServerCallInfo<ReqT, RespT> {
+    private final MethodDescriptor<ReqT, RespT> methodDescriptor;
+    private final Attributes attributes;
+    private final String authority;
+
+    ServerCallInfo(
```

Comment:
Hmm... if you lock down the constructor, how do people outside of gRPC unit-test their own ServerStreamTracers?

Maybe this should be an abstract class, and put the impl in `io/grpc/internal`.